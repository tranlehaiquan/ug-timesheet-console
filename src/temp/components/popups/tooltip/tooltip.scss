@import '../../../../variables.scss';
@import '../popup.scss';

$tooltip-max-width: 360px;
$tooltip-default-height: 32px;
$tooltip-dark-background-color: $midnight-blue-1 !default;
$tooltip-light-background-color: $white-1 !default;
$tooltip-color: $white-1 !default;
$tooltip-color-inverse: $black-1;
$tooltip-padding: 9px 12px 7px !default;
$tooltip-line-height: 1.2 !default;
$tooltip-letter-spacing: 0.2px !default;

/* purgecss start ignore */
.sked-tooltip {
  .sked-infowindow__content {
    box-sizing: border-box;
    padding: $tooltip-padding;
    border: none;
    box-shadow: $tooltip-box-shadow;

    max-width: $tooltip-max-width !important;
    min-width: 100px;
    min-height: $tooltip-default-height;
    text-align: center;
    line-height: $tooltip-line-height;
    letter-spacing: $tooltip-letter-spacing;
  }

  &--dark {
    .sked-infowindow__content {
      color: $tooltip-color;
      background: $tooltip-dark-background-color;
    }

    .sked-infowindow__triangle {
      background: $tooltip-dark-background-color;
    }
  }

  &--light {
    .sked-infowindow__content {
      color: $tooltip-color-inverse;
      background: $tooltip-light-background-color;
    }

    .sked-infowindow__triangle {
      background: $tooltip-light-background-color;
    }
  }
}
/* purgecss end ignore */
