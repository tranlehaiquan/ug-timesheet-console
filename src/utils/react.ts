import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
