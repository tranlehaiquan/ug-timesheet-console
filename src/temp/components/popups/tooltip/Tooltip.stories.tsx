import * as React from "react";
import { storiesOf } from "@storybook/react";
import { select } from "@storybook/addon-knobs/react";

import { Button } from "../../buttons/button/Button";
import { Tooltip } from "./Tooltip";

storiesOf("Tooltip", module).add("Basic", () => {
  const colorScheme = select("Colour Scheme", ["dark", "light"], "dark");
  return (
    <Tooltip
      content={"Tooltip content"}
      position={"top"}
      colorScheme={colorScheme}
    >
      <Button buttonType="secondary">Hover Trigger</Button>
    </Tooltip>
  );
});
