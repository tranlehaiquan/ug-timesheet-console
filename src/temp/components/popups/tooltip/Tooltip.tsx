import * as React from "react";
import { Position } from "../info-window/info-window-utils";

import {
  InfoWindow,
  IInfoWindowProps,
} from "../info-window/InfoWindow";

import "./tooltip.scss";

export type ITooltipPosition = Position;
type IColorScheme = "dark" | "light";

export interface ITooltipProps extends IInfoWindowProps {
  /**
   * Colour scheme for the tooltip content.  Available options are 'dark' and 'light'
   */
  colorScheme?: IColorScheme;
}

/**
 * Tooltips are an extened version of the info window that triggers on hover by default, delayed slightly and is styled with tooltip styles
 */
export class Tooltip extends React.PureComponent<React.PropsWithChildren<ITooltipProps>, {}> {
  static defaultProps = {
    event: "hover",
    delayShow: 250,
    colorScheme: "dark",
  };

  render() {
    const { colorScheme, children, containerClassName } = this.props;

    const containerClasses = `sked-tooltip sked-tooltip--${colorScheme} ${
      containerClassName || ""
    }`;
    return (
      <InfoWindow {...this.props} containerClassName={containerClasses}>
        {children}
      </InfoWindow>
    );
  }
}
