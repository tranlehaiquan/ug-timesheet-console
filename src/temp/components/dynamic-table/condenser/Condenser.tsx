import * as React from "react";
import { Tooltip } from "../../popups/tooltip/Tooltip";

export interface ICondenserProps {
  children?: string | React.ReactNode;
}

export const Condenser: React.FC<ICondenserProps> = ({ children }) => {
  const ref = React.useRef();
  const [overflow, setOverflow] = React.useState(false);

  const checkOverflow = () => {
    if (ref && ref.current) {
      const wrapper: HTMLElement = ref.current;
      setOverflow(wrapper.clientWidth < wrapper.scrollWidth);
    }
  };

  return (
    <Tooltip
      preventShow={!overflow}
      content={children}
      delayShow={250}
      position="top"
      className="sk-truncate"
    >
      <div
        data-sk-name="children-container"
        ref={ref}
        className="sk-truncate"
        onMouseEnter={checkOverflow}
      >
        {children}
      </div>
    </Tooltip>
  );
};
