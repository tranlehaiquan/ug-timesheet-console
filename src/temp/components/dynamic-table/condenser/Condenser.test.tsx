import * as React from 'react'
import { Condenser } from './Condenser'
import { mount, render } from 'enzyme'
import { Tooltip } from '../../popups/tooltip/Tooltip'

const selectors = {
  childrenContainer: '[data-sk-name="children-container"]'
}

const getSampleCondenser: (children?: string | React.ReactNode) => React.ReactElement = (children = '') => (
  <Condenser>
    { children }
  </Condenser>
)

describe('Condenser', () => {

  test('with string label', () => {
    const label = 'sample-label'
    const condenser = mount(getSampleCondenser(label))
    const childrenContainer = condenser.find(selectors.childrenContainer)
    expect(childrenContainer.text()).toEqual(label)
  })

  test('with child component', () => {
    const condenser = render(getSampleCondenser((
      <>
        <p>child1</p>
        <p>child2</p>
      </>
    )))
    expect(condenser).toMatchSnapshot()
  })

  test('correct tooltip settings', () => {
    const condenser = mount(getSampleCondenser())
    const tooltip = condenser.find(Tooltip)
    expect(tooltip.length).toEqual(1)
    expect(tooltip.props().position).toEqual('top')
    expect(tooltip.props().className).toEqual('sk-truncate')
    expect(tooltip.props().delayShow).toEqual(250)
    expect(tooltip.props().preventShow).toEqual(true)
  })

})
