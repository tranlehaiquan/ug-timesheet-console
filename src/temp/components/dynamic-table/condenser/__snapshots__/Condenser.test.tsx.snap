// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condenser with child component 1`] = `
<div
  class="sked-infowindow__trigger sk-truncate"
  data-sk-name="sked-infowindow-trigger"
>
  <div
    class="sk-truncate"
    data-sk-name="children-container"
  >
    <p>
      child1
    </p>
    <p>
      child2
    </p>
  </div>
</div>
`;
